gnovm/stdlibs/std/emit_event.go

```diff
@@ -46,12 +59,20 @@ func attrKeysAndValues(attrs []string) ([]gnoEventAttribute, error) {
 }
 
 type gnoEvent struct {
-	Type       string              `json:"type"`
+	OrigCaller string              `json:"orig_caller"`
+	Prev       prevStack           `json:"prev"`
+	PrevFunc   string              `json:"prev_func"`
```

Comment:
I would remove this, or at most put it together with the prevStack struct.

Reply:
```go
type prevStack struct {
  PrevPkgPath string `json:"prev_pkg_path"`
  PrevPkgAddr string `json:"prev_pkg_addr"`
  PrevFunc    string `json:"prev_func"`
}
```

If you're suggesting something like this, there is little problem.

Because of same prefix `Prev`, user might think like theses all 3 prevs are from same pkg_path (or same place, same level).

But as test cases from here, prevFunc from `SubFoo()` will be `Foo()` which is at same pkg_path ( not likely prev path )
https://github.com/gnolang/gno/blob/252577668f81e62062e24f63f2feca98ea60dccc/gno.land/cmd/gnoland/testdata/event_normal.txtar#L33-L45

What do you think of this?

Reply:
Actually, can we remove PrevFunc?

From what I understand, there is no guarantee that this isn't in the same package. I've looked at the examples and I don't understand what it is (it is not the caller's caller, but the caller's caller's caller?!), and it doesn't seem correct to put a function name here when it doesn't have an associated PkgPath and Type. (what if it's a method? or comes from another package? how would I know? etc.)

I'd argue that even adding `Func` in the first place is misguided, and that having a PkgPath of the current realm would be enough, but I digress. I may do another PR after this one.

Reply:
`prevFunc` is prev function of function that called Emit(). 
It does make a quite confuse even I'm explaining what it is right now :( 
- Call1() -> Call2() - std.Emit() // at this point Call1 will be prevFunc

---

> From what I understand, there is no guarantee that this isn't in the same package. ~~~ and it doesn't seem correct to put a function name here when it doesn't have an associated PkgPath and Type

This is exact reason for why I divided `prev`, `prevF
unc`, and `pkgPath`. They might not come from same packages.

---
> I'd argue that even adding Func in the first place is misguided,

For an observability, current `Func` and `PkgPath` looks good to me though.